shor/gates.py

```diff
@@ -31,8 +42,23 @@ def num_states(self):
     def to_matrix(self) -> np.ndarray:
         return np.eye(self.num_states())
 
+    @property
+    def matrix(self):
+        return self.to_matrix()
+
+    def invert(self):
+        return self
```

Comment:
Why return self?

Reply:
More a placeholder than anything else.

Gates are their own unitary inverse, but was marking a spot in the API since we utilize inverse QFT for shor's algorithm.

I also noticed this in another API and thought it could be useful.